services/app-api/src/resolvers/healthPlanPackage/updateHealthPlanFormData.ts

```diff
@@ -59,168 +114,277 @@ export function updateHealthPlanFormDataResolver(
+                    extensions: {
+                        code: 'INTERNAL_SERVER_ERROR',
+                        cause: 'DB_ERROR',
+                    },
+                })
+            }
+
+            // Authorize the update
+            const stateFromCurrentUser = context.user.stateCode
+            if (contractWithHistory.stateCode !== stateFromCurrentUser) {
+                logError(
+                    'updateHealthPlanFormData',
+                    'user not authorized to fetch data from a different state'
+                )
+                setErrorAttributesOnActiveSpan(
+                    'user not authorized to fetch data from a different state',
+                    span
+                )
+                throw new ForbiddenError(
+                    'user not authorized to fetch data from a different state'
+                )
+            }
+
+            // Can't update a submission that is locked or resubmitted
+            if (!['DRAFT', 'UNLOCKED'].includes(contractWithHistory.status)) {
+                const errMessage = `Package is not in editable state: ${input.pkgID} status: ${contractWithHistory.status}`
+                logError('updateHealthPlanFormData', errMessage)
+                setErrorAttributesOnActiveSpan(errMessage, span)
+                throw new UserInputError(errMessage, {
+                    argumentName: 'pkg',
+                })
+            }
+
+            // Check if draft revision exist, if not then return error
+            if (!contractWithHistory.draftRevision) {
+                const errMessage = `Issue finding a draft revision for contract id ${input.pkgID}. Message: Draft revision not found}`
+                throw new GraphQLError(errMessage, {
+                    extensions: {
+                        code: 'NOT_FOUND',
+                        cause: 'DB_ERROR',
+                    },
+                })
+            }
+
+            //check for fields
+            // Validate that none of these protected fields have been modified.
+            // These fields should only be modified by the server, never by an update.
+            const previousFormData = {
+                id: contractWithHistory.draftRevision.id,
+                stateCode: contractWithHistory.stateCode,
+                stateNumber: contractWithHistory.stateNumber,
+                // Matches the format of the created at date that comes out of the proto.
+                createdAt: new Date(
+                    Date.UTC(
+                        contractWithHistory.draftRevision.updatedAt.getUTCFullYear(),
+                        contractWithHistory.draftRevision.updatedAt.getUTCMonth(),
+                        contractWithHistory.draftRevision.updatedAt.getUTCDate()
+                    )
+                ),
+                updatedAt: contractWithHistory.draftRevision.updatedAt,
+            }
+
+            const unfixedFields = validateProtectedFields(
```

Comment:
Do we even need to handle this case for the feature flag on case? I would assume we just ignore even if the proto is adjusting fields that can't be edited - because we aren't actuall writing those fields to the data and are relying on prisma handler to update things reliably (or error handle is possible concurrent calls are happening)

Reply:
My thought was in case anyone of these fields did drift, we wouldn't want to update the data. For, example when two users try to update the same submission at the same time, one of them would be working with stale data.

Reply:
Yes I hope we can drop this, if we want to check on the updated at thing we can make that a separate error that indicates that concurrent editing is happening, but none of the rest of those fields should be settable from the API anymore.